### PR TITLE
bring Heroku's ember example up to date with modern Ember

### PR DESCRIPTION
**Pull request by jm3**
- update npm deps to latest versions
- lock dependency tree versions using yarn (faster installs, too)
- rm bower (now unused as all ember pkgs are in `package.json`)
- move ember deps from bower to npm: per **leizhao4** in #-ember-cli Slack:
> jm3 Since 2.11 you should pull `ember-source` from npm instead of `ember` from bower. Also `ember-cli-shims` and `ember-cli-test-loader`.

site renders and deploys the same as before :)

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,9 +0,0 @@
-{
```
Comment (Dhaulagiri) [inline bower.json]: until https://github.com/heroku/heroku-buildpack-emberjs/issues/24 is addressed we will still need an empty version of this file to exist in the project for the buildpack to deploy the app successfully.  You can leave the existing file in place and just remove the `dependencies` section.

**Review by Dhaulagiri — CHANGES_REQUESTED** [review]
Thanks for doing this! There is one small change we need to make that I commented on otherwise this looks good to me.

File: bower.json (hunk 1 above)
Comment (Dhaulagiri) [reply bower.json]: @jm3 your comment seems to indicate that you might have been able to get this branch to deploy? To sanity check myself I checked out your branch and deployed to a new app and got the error I have seen with other similar setups:

```
remote: -----> Installing bower dependencies
remote: bower                           ENOENT No bower.json present
remote: Error running: bower --allow-root install 2>&1
remote: -----> Failed trying to compile heroku/ember-cli-deploy
remote:  !     Push rejected, failed to compile emberjs app.
remote: 
remote:  !     Push failed
remote: Verifying deploy....
remote: 
remote: !	Push rejected to thawing-eyrie-85869.
```

File: README.md [hunk 2]
```diff
@@ -20,7 +19,6 @@ You will need the following things properly installed on your computer.
 $ git clone git@github.com:heroku/ember-js-getting-started.git
 $ cd ember-js-getting-started
 $ npm install
```
Comment (Dhaulagiri) [inline README.md]: could you update the instructions here to use yarn instead of npm? (and probably add yarn as a dependency in the list above)

**Comment by caseywatts** [comment]
@jm3 I wanna thank you for doing this PR too! :D
🎉 🎉

File: README.md (hunk 2 above)
Comment (jm3) [reply README.md]: happy to!

**Comment by jm3** [comment]
@Dhaulagiri readme updated to prefer yarn over npm, bower.json restored without dependencies. 

Deploying works fine: [calm-waters-35963](https://calm-waters-35963.herokuapp.com/)!

**Review by Dhaulagiri — APPROVED** [review]

**Comment by Dhaulagiri** [comment]
Thanks!

**Pull request merged**